### PR TITLE
03-1125: (bug) fix current visit retaining previous patient visit-data

### PR DESCRIPTION
**Pull request by donaldkibet**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

- Fix bug where, on searching a different patient, the previous patient `visit` data would still be displayed as the patient's visit. This is seen on the `active-visit-tag.component` on patient chart

## Screenshots

current-visit

## Issue

[https://issues.openmrs.org/projects/MF/issues/O3-1125](https://issues.openmrs.org/projects/MF/issues/O3-1125)

## Other

This is related to  [https://github.com/openmrs/openmrs-esm-core/pull/332](https://github.com/openmrs/openmrs-esm-core/pull/332)

### CODE REVIEW

File: packages/esm-patient-banner-app/src/banner-tags/active-visit-tag.component.tsx [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
+import { InlineLoading, Tag, TooltipDefinition } from 'carbon-components-react';
 import { formatDatetime, parseDate, useVisit } from '@openmrs/esm-framework';
 interface ActiveVisitBannerTagProps {
   patientUuid: string;
   patient: fhir.Patient;
 }
 const ActiveVisitBannerTag: React.FC<ActiveVisitBannerTagProps> = ({ patientUuid, patient }) => {
   const { t } = useTranslation();
-  const { currentVisit } = useVisit(patientUuid);
+  // @ts-ignore
```
Comment (donaldkibet) [inline packages/esm-patient-banner-app/src/banner-tags/active-visit-tag.component.tsx:12]: To be remove once [https://github.com/openmrs/openmrs-esm-core/pull/332](https://github.com/openmrs/openmrs-esm-core/pull/332) is merged in

File: packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx [hunk 2]
```diff
@@ -52,6 +50,8 @@ const StartVisitForm: React.FC<DefaultWorkspaceProps> = ({ patientUuid, closeWor
   const [visitTime, setVisitTime] = useState(dayjs(new Date()).format('hh:mm'));
   const [visitType, setVisitType] = useState<string | null>(null);
   const state = useMemo(() => ({ patientUuid }), [patientUuid]);
+  // @ts-ignore
+  const { updateCurrentVisit } = useVisit();
```
Comment (brandones) [inline packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx:54]: How is `useVisit` supposed to work with no `patientUuid` argument? The function is typed to require `patientUuid`, and seems written to require it. Perhaps `updateCurrentVisit` should just be independent of `useVisit`?

**Comment by brandones** [comment]
Looks like the build needs to be fixed, @donaldkibet . Maybe just need to update the core dependency?

**Comment by brandones** [comment]
Ping @donaldkibet ?

**Comment by donaldkibet** [comment]
This [PR](https://github.com/openmrs/openmrs-esm-core/commit/a8176d8435ed122c30e9264a1c2c3039bb2275fe) broke the CI, for core. will need to resolve that first then merge this in

**Pull request merged**